File: packages/core/src/scanner/Chunks.ts
export type ChunkKind = 'slidingWindow' | 'wholeFile' | 'selection';

export interface RuleHint {
  signalId: string;
  line: number;
  snippet: string;
}

export interface CodeChunk {
  id: string;
  file: string;
  startLine: number;
  endLine: number;
  kind: ChunkKind;
  code: string;
  ruleHints: RuleHint[];
}

File: packages/intellij-plugin/src/main/kotlin/ai/astha/scantheplanet/idea/scanner/Chunks.kt
package ai.astha.scantheplanet.idea.scanner

import java.nio.file.Path

enum class ChunkKind {
    SlidingWindow,
    WholeFile,
    Selection
}

data class RuleHint(
    val signalId: String,
    val line: Int,
    val snippet: String
)

data class CodeChunk(
    val id: String,
    val file: Path,
    val startLine: Int,
    val endLine: Int,
    val kind: ChunkKind,
    val code: String,
    var ruleHints: List<RuleHint> = emptyList()
)
